File: src/modifiers/antialiased.ts
import { createModifier } from './createModifier';

/// SwiftUI `.antialiased(_:)`. Image-only — smooths edges when the
/// image is drawn rotated, scaled, or otherwise off-pixel-grid.
export function antialiased(value: boolean = true) {
  return createModifier('antialiased', { value });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/Antialiased.swift
import SwiftUI

/// SwiftUI `.antialiased(_:)` — Image-only. Turns on edge smoothing
/// when the image is drawn rotated, scaled, or otherwise off-grid.
enum RNWAntialiasedModifier {
    @MainActor
    static func register(into r: RNWImageModifierRegistry) {
        r.register("antialiased") { image, params in
            image.antialiased(params.bool("value") ?? true)
        }
    }
}
